tests/robots/test_robot.py

```diff
@@ -62,5 +68,93 @@ def test_basic_name_links(ur5_links):
 
 
 def test_basic_attr():
-    robot = Robot.basic('testbot', location="rfl")
-    assert robot.model.attr['location'] == "rfl"
+    robot = Robot.basic('testbot', location='rfl')
+    assert robot.model.attr['location'] == 'rfl'
+
+
+def test_name(panda_robot_instance):
+    robot = panda_robot_instance
+    assert robot.name == 'panda'
+
+
+def test_group_names(ur5_robot_instance):
+    robot = ur5_robot_instance
+    assert robot.group_names == ['manipulator', 'endeffector']
+
+
+def test_main_group_name(panda_robot_instance):
+    robot = panda_robot_instance
+    assert robot.main_group_name == 'panda_arm_hand'
+
+
+def test_root_name(ur5_robot_instance):
+    robot = ur5_robot_instance
+    assert robot.root_name == 'world'
+
+
+@pytest.mark.parametrize('group,expected_result', [(None, 'panda_rightfinger'), ('panda_arm_hand', 'panda_rightfinger'), ('panda_arm', 'panda_link8')])
+def test_get_end_effector_link_name(panda_robot_instance, group,  expected_result):
+    robot = panda_robot_instance
+    assert robot.get_end_effector_link_name(group=group) == expected_result
+
+
+def test_get_end_effector_link_name_wrong_group(panda_robot_instance):
+    robot = panda_robot_instance
+    with pytest.raises(KeyError):
+        robot.get_end_effector_link_name(group='panda_leg')
+
+
+@pytest.mark.parametrize('group', [None, 'endeffector'])
+def test_get_end_effector_link(ur5_robot_instance, group):
+    robot = ur5_robot_instance
+    assert isinstance(robot.get_end_effector_link(group=group), compas.robots.Link)
+
+
+@pytest.mark.parametrize('group, x_coord', [(None, .301), ('panda_arm', .359)])
+def test_get_end_effector_frame(panda_robot_instance, group, x_coord):
+    robot = panda_robot_instance
+    ee_frame = robot.get_end_effector_frame(group=group)
+    assert round(ee_frame.point.x, 3) == x_coord
+
+
+@pytest.mark.parametrize('group, remove_semantics, expected_result',
+                         [(None,          False, 'base_link'),
+                          (None,          True,  'base_link'),
+                          ('endeffector', False, 'ee_link'),
+                          ('endeffector', True,  'base_link')])
+def test_get_base_link_name(ur5_robot_instance, group, remove_semantics, expected_result):
```

Comment:
I think this test is too complicated. Tests two code paths that are very different.
If you want to test with and without semantics, I would write two different tests (eg `test_get_base_link_name` and `test_get_base_link_name_without_semantics`), and for the without_semantics, make sure you create one without semantics from the start, not removing the semantics after being created with one.